### PR TITLE
Flichtenheld/jenkins windows agent

### PR DESCRIPTION
**Pull request by flichtenheld**
Trying to resuse the work that went into buildbot EC2 for my Jenkins agents.

Plus some general improvements, like:

- add some error checking
- fix reboot handling
- add variable for subnet

### CODE REVIEW

File: buildbot-host/scripts/build-deps.ps1 [hunk 1]
```diff
@@ -1,45 +1,38 @@
...
+. C:\Windows\Temp\ps_support.ps1
+
 Write-Host "Setting up openvpn build dependencies with vcpkg"
 
 if (-Not (Test-Path $workdir)) {
   New-Item -Type directory $workdir
+  CheckLastExitCode
 }
 
+Add-MpPreference -ExclusionPath $workdir
```
Comment (mattock) [inline buildbot-host/scripts/build-deps.ps1:12]: Interesting, never had to use this. But if it is needed in some cases, so be it :smile:.

**Comment by mattock** [comment]
In general looks good and I could not spot anything obviously wrong in there. Documentation for the jenkinsmaster_address parameter is missing though, and documenting how to build just one image (jenkins or buildbot) would be nice. I'm testing the code for buildbot regressions now - that will take some hours probably.

That said, I would like to improve the hygiene of the repository by restructuring the repository along these lines:

```
openvpn-buildbot/
├── buildbot-host
│   └── buildbot-worker-windows-server-2019
├── jenkins
│   └── jenkins-agent-windows-server-2019
└── scripts
```

Packer-vise this new layout should work ok without duplicating much code. The parameter names in Packer could also be made specific to buildbot and jenkins. Docker should also be fine with this with minor path changes.

This layout would probably make it easier to work together on other bootstrapping scripts. For example those used to spin up buildbot workers and jenkins agents for Linux. Besides the Dockerfiles there only one such script I hastily set up for cron2 (_buildbot-host/scripts/install-openvpn-build-deps-ubuntu.sh_), but more will follow at some point.

I think this layout would also make it easier to share scripts (or Dockerfiles?) that set up build requirements for openvpn, openvpn3, openvpn3-linux, etc.

Thoughts?

File: buildbot-host/scripts/build-deps.ps1 (hunk 1 above)
Comment (flichtenheld) [reply buildbot-host/scripts/build-deps.ps1:12]: It is not strictly required. It just means that the machine will spend a significant amount of its CPU capacity on running AV. If you have 2 CPUs, I usually see around 50% CPU going to AV during builds. If you have more cores, then it matters less. I will definitely leave it in the Jenkins code.

**Comment by flichtenheld** [comment]
> Thoughts?

So I thought we could reduce code duplication by putting both builds in the same packer file. But in the end the result doesn't really save all that much because it did decide that a separate build is required and not just two different sources in the same build.

So yeah, no objections.

**Comment by mattock** [comment]
I'll refactor the code then. I'll use this work as the basis and try to keep Jenkins part unbroken.

**Comment by mattock** [comment]
I moved the stuff around in #2. It is based on this PR. Buildbot side seems to work ok, did not test the Jenkins side but with some luck it does work.

**Pull request merged**